CONTRIBUTING.md

```diff
@@ -0,0 +1,113 @@
+## How to Contribute
+
+The OCS-Operator project is under the [Apache 2.0 license](LICENSE). We accept
+contributions via GitHub pull requests. This document outlines how to
+contribute to the project.
+
+## Contribution Flow
+
+Developers must follow these steps to make a change:
+
+1. Fork the `openshift/ocs-operator` repository on GitHub.
+2. Create a branch from the master branch, or from a versioned branch (such
+   as `release-4.2`) if you are proposing a backport.
+3. Make changes.
+4. Create tests as needed and ensure that all tests pass.
+5. Ensure your commit messages include sign-off.
+6. Push your changes to a branch in your fork of the repository.
+7. Submit a pull request to the `openshift/ocs-operator` repository.
+8. Work with the community to make any necessary changes through the code
+   review process (effectively repeating steps 3-8 as needed).
+
+## Developer Environment Installation
+
+Instructions to create a dev environment for OCS-operator can be found in the
+[main project documentation](./README.md#Installation of development builds).
+
+## Commit Requirements
+
+In addition to passing [functional tests](./README.md#functional-tests), all
+commits to OCS-operator must follow the guidelines in this section. You can
+verify your changes meet most of these standards by running the following
+command in the repository's root directory:
+
+```
+make ocs-operator-ci
+```
+
+### Commits Per Pull Request
+
+*This requirement cannot be tested by make ocs-operator-ci.*
+
+OCS-operator is a project which maintains several versioned branches
+independently. When backports are necessary, monolithic commits make it
+difficult for maintainers to cleanly backport only the necessary changes.
+
+Where possible, pull requests should be composed of multiple commits that
+each make small but meaningful changes. Striking a balance between minimal
+commits and logically complete changes is an art as much as a science, but
+when it is possible and reasonable, divide your pull request into more commits.
+
+### Commit and Pull Request Messages
+
+*This requirement cannot be fully tested by make ocs-operator-ci.*
+
+All commit messages must contain:
+- A single summary line describing the change
+- A single blank line
+- A longer description of the change, including:
```

Comment:
This can be optional if the first line says it all.

Reply:
In my experience, developers will tend to believe that the first line says it all (much the way developers tend to universally believe that their code is self-documenting). And even if the summary does fully explain _what_ the change does, it almost never says _why_ the change was needed, which is very useful information for codebase maintainers.

Still, If we want me to make this optional, I certainly can.

Reply:
It's optional but recommended. For example, in a commit to fix one or a few typos in comments I wouldn't ask for anything other than the summary line.

Reply:
Let's agree to make it optional, but highly recommended.